cloud-controller-manager/do/loadbalancers.go

```diff
@@ -275,6 +274,57 @@ func (l *loadBalancers) EnsureLoadBalancer(ctx context.Context, clusterName stri
 	}, nil
 }
 
+func (l *loadBalancers) updateLoadBalancer(ctx context.Context, lb *godo.LoadBalancer, lbRequest *godo.LoadBalancerRequest, service *v1.Service) error {
+	var lbCertID string
+	for _, rule := range lb.ForwardingRules {
+		if rule.TlsPassthrough {
+			continue
+		}
+		if rule.CertificateID != "" {
+			lbCertID = rule.CertificateID
+			// it's not clear that it is even possible for different forwarding rules
+			// on the same loadbalancer to have different certificates associated
+			// with them -- as far as we are concerned where a load balancer is
+			// created by CCM in the first place it doesn't seem possible so we only
+			// need to grab the first one we find
+			break
+		}
+	}
+
+	checkServiceCertID := true
+	serviceCertID := getCertificateID(service)
+	if lbCertID != "" && lbCertID != serviceCertID {
+		lbCert, _, err := l.resources.gclient.Certificates.Get(ctx, lbCertID)
+		if err != nil {
+			return fmt.Errorf("failed to get current certificate for lb")
+		}
+
+		if lbCert.Type == "lets_encrypt" {
+			l.ensureCertificateIDAnnot(service, lbCertID)
```

Comment:
Should this set a the new `serviceCertID = getCertificateID(service)` here too?

Reply:
It's not strictly necessary; at the line just below this we set `checkServiceCertID` to `false` and the block conditioned on that variable is the only place where the service cert id is used.

That being said, I can see how someone else coming at this code later on might be confused so it probably does make sense to set it to over write the old value with that of `lbCertID`.

Reply:
Alternatively, it `getCertificateID` is a fairly cheap call that just indexes into a map to return a value...while I know next to nothing about the go compiler or compilers in general, i wouldn't be surprised to learn functions like that are simply inlined... [the doc on go compiler optimizations suggests this is probably the case](https://github.com/golang/go/wiki/CompilerOptimizations#function-inlining)

So where I'm going with this is, I think I'll just replace all `serviceCertID` instances with the function call and avoid the potential for confusion altogether.

Reply:
The extra lookup is certainly not going to affect us performance-wise.